src/kernels/kernel.node.ts

```diff
@@ -2,171 +2,75 @@
-        return this.kernelExecution.queue;
-    }
-    private _disposed?: boolean;
-    private _disposing?: boolean;
-    private _ignoreNotebookDisposedErrors?: boolean;
-    private readonly _kernelSocket = new ReplaySubject<KernelSocketInformation | undefined>();
-    private readonly _onStatusChanged = new EventEmitter<KernelMessage.Status>();
-    private readonly _onRestarted = new EventEmitter<void>();
-    private readonly _onStarted = new EventEmitter<void>();
-    private readonly _onDisposed = new EventEmitter<void>();
-    private readonly _onPreExecute = new EventEmitter<NotebookCell>();
-    private _notebookPromise?: Promise<INotebook>;
-    private readonly hookedNotebookForEvents = new WeakSet<INotebook>();
-    private eventHooks: ((ev: 'willInterrupt' | 'willRestart') => Promise<void>)[] = [];
-    private restarting?: Deferred<void>;
+export class Kernel extends BaseKernel {
     private readonly kernelExecution: KernelExecution;
     private disposingPromise?: Promise<void>;
-    private startCancellation = new CancellationTokenSource();
-    private startupUI = new DisplayOptions(true);
     constructor(
-        public readonly id: Uri,
-        public readonly resourceUri: Resource,
-        public readonly kernelConnectionMetadata: Readonly<KernelConnectionMetadata>,
-        private readonly notebookProvider: INotebookProvider,
-        private readonly disposables: IDisposableRegistry,
-        private readonly launchTimeout: number,
+        id: Uri,
+        resourceUri: Resource,
+        kernelConnectionMetadata: Readonly<KernelConnectionMetadata>,
+        notebookProvider: INotebookProvider,
+        disposables: IDisposableRegistry,
+        launchTimeout: number,
         interruptTimeout: number,
-        private readonly appShell: IApplicationShell,
+        appShell: IApplicationShell,
         private readonly fs: IFileSystem,
-        public readonly controller: NotebookController,
-        private readonly configService: IConfigurationService,
+        controller: NotebookController,
+        configService: IConfigurationService,
         outputTracker: CellOutputDisplayIdTracker,
-        private readonly workspaceService: IWorkspaceService,
-        readonly cellHashProviderFactory: CellHashProviderFactory,
+        cellHashProviderFactory: CellHashProviderFactory,
+        workspaceService: IWorkspaceService,
         private readonly pythonExecutionFactory: IPythonExecutionFactory,
-        private statusProvider: IStatusProvider
+        private readonly statusProvider: IStatusProvider
     ) {
+        super(
+            id,
+            resourceUri,
+            kernelConnectionMetadata,
+            notebookProvider,
+            disposables,
+            launchTimeout,
+            appShell,
+            controller,
+            configService,
+            workspaceService,
+            cellHashProviderFactory
+        );
         this.kernelExecution = new KernelExecution(
```

Comment:
KernelExecution is node specific at the moment, so chopped off this work at this point.

When I get to implementing the kernel for web, KernelExecution will likely be the starting point.